tools/ci_credentials/ci_credentials/secrets_loader.py

```diff
@@ -0,0 +1,240 @@
+import base64
+import json
+import re
+from dataclasses import dataclass
+from json.decoder import JSONDecodeError
+from pathlib import Path
+from typing import Mapping, Any, Tuple, ClassVar
+
+from ci_common_utils import GoogleApi
+from ci_common_utils import Logger
+
+DEFAULT_SECRET_FILE = "config"
+DEFAULT_SECRET_FILE_WITH_EXT = DEFAULT_SECRET_FILE + ".json"
+
+GSM_SCOPES = ("https://www.googleapis.com/auth/cloud-platform",)
+
+
+@dataclass
+class SecretsLoader:
+    """
+    """
+    logger: ClassVar[Logger] = Logger()
+    # base_folder: ClassVar[Path] = Path("/Users/pixel/Projects/Airbyte/repo/")
+    base_folder: ClassVar[Path] = Path("/actions-runner/_work/airbyte/airbyte/")
+
+    gsm_credentials: Mapping[str, Any]
+    github_secrets: Mapping[str, str]
+
+    connector_name: str = None
+    _api: GoogleApi = None
+
+    @classmethod
+    def get_github_registers_filepath(cls) -> Path:
+
+        return cls.base_folder / "tools/bin/ci_credentials.sh"
+
+    def __read_github_registers_file(self) -> Mapping[Tuple[str, str], str]:
+        """
+            This function is used as workaround  for migration script.
+            And it should be remove after final migration
+        """
+        registers_filepath = self.get_github_registers_filepath()
+        if not registers_filepath.exists():
+            # it is optional logic and we can skip sync if a GitHub secrets' register is not exist
```

Comment:
if we're removing Github secret usage, isn't the code that is getting github secrets redundant?

Reply:
I've removed all github secrets' logic fully